app/javascript/admin/controllers/confirmation_modal_controller.js

```diff
@@ -13,20 +13,38 @@ window.addEventListener('load', () => {
   }
 });
 
+const nonRedirectEndpoints = [
```

Comment:
I definitely like this better than the `switch` we had before. I'm slightly worried about `routes` getting out of sync but I feel that's something we should tackle more comprehensively, not as part of this PR.